ui/src/app/(dashboard)/newstream/FunctionList.tsx

```diff
@@ -1,4 +1,31 @@
-import statusStyles from '@/utils/statusStyles';
+import {
+  IconStatusCircleCheck,
+  IconStatusCircleArrowPath,
+  IconStatusCircleCross,
+  IconStatusCircleMinus,
+} from '@/icons';
+import { FunctionRunStatus } from '@/store/generated';
+
+export function statusStyles(status: FunctionRunStatus | null) {
+  switch (status) {
+    case FunctionRunStatus.Running:
+      return {
+        icon: IconStatusCircleArrowPath,
+      };
+    case FunctionRunStatus.Completed:
+      return {
+        icon: IconStatusCircleCheck,
+      };
+    case FunctionRunStatus.Failed:
+      return {
+        icon: IconStatusCircleCross,
+      };
+    default:
+      return {
+        icon: IconStatusCircleMinus,
+      };
+  }
+}
```

Comment:
**suggestion (non-blocking)**: this can be simplified by using a map:

```suggestion
const functionRunStatusIcons = {
  [FunctionRunStatus.Running]: IconStatusCircleArrowPath,
  [FunctionRunStatus.Completed]: IconStatusCircleCheck,
  [FunctionRunStatus.Failed]: IconStatusCircleCross,
  [FunctionRunStatus.Cancelled]: IconStatusCircleMinus,
} as const satisfies Record<FunctionRunStatus, React.ComponentType>;
```